module-apps/apps-common/ApplicationCommonPopupBlueprints.cpp

```diff
@@ -49,14 +49,15 @@ namespace app
                          std::to_string(volumeParams->getVolume()).c_str());
                 auto volume          = volumeParams->getVolume();
                 auto context         = volumeParams->getAudioContext();
+                auto source          = volumeParams->getRequestSource();
                 const auto popupName = resolveWindowName(gui::popup::ID::Volume);
                 if (const auto currentWindowName = getCurrentWindow()->getName(); currentWindowName == popupName) {
-                    updateCurrentWindow(std::make_unique<gui::VolumePopupData>(volume, context));
+                    updateCurrentWindow(std::make_unique<gui::VolumePopupData>(volume, context, source));
```

Comment:
Why not move the construction of the popup in front of the condition to keep the code DRY.

Reply:
Sure, fixed!